pkg/util/log/log.go

```diff
@@ -36,6 +37,8 @@ var (
 	RXTolerantResourceID = regexp.MustCompile(`(?i)^(?:/admin)?/subscriptions/([^/]+)(?:/resourceGroups/([^/]+)(?:/providers/([^/]+)/([^/]+)(?:/([^/]+))?)?)?`)
 
 	RXTolerantSubResourceID = regexp.MustCompile(`(?i)^(?:/admin)?/subscriptions/([^/]+)(?:/resourceGroups/([^/]+)(?:/providers/([^/]+)/([^/]+)/([^/]+)(?:/([^/]+))?)?)?`)
+
+	cachedLog *logrus.Entry
```

Comment:
This make logger global. This is anti-pattern in golang. Usually logger should be passed down so component could append he structure if needed so keeping original structure with enriched data.
By doing this we are at risk somebody somewhere in 6 month time overrides this by not knowingly calling `getBaseLogger` and all our logging is tossed. 

I would better try to avoid situation like this where we need to know these "knowhows" and look for them in code reviews.